src/request.ts

```diff
@@ -1186,6 +1187,40 @@ class DatastoreRequest {
     this.save(entities, callback);
   }
 
+  merge(entities: Entities): Promise<CommitResponse>;
+  merge(entities: Entities, callback: CallOptions): void;
+  /**
+   * Maps to {@link Datastore#save}, forcing the method to be `merge`.
+   *
+   * @param {object|object[]} entities Datastore key object(s).
+   * @param {Key} entities.key Datastore key object.
+   * @param {string[]} [entities.excludeFromIndexes] Exclude properties from
+   *     indexing using a simple JSON path notation. See the examples in
+   *     {@link Datastore#save} to see how to target properties at different
+   *     levels of nesting within your entity.
+   * @param {object} entities.data Data to merge to the same for provided key.
+   * @param {function} callback The callback function.
+   * @param {?error} callback.err An error returned while making this request
+   * @param {object} callback.apiResponse The full API response.
+   */
+  merge(
+    entities: Entities,
+    callback?: CallOptions
+  ): void | Promise<CommitResponse> {
+    arrify(entities)
+    .map(DatastoreRequest.prepareEntityObject_)
+    .map((x: PrepareEntityObjectResponse, index: number, array: PrepareEntityObjectResponse[]) => {
+      this.get(x.key, (err: ServiceError, data: Entity) => {
+        x.method = 'upsert';
+        x.data = Object.assign({}, data, x.data);
+        array[index] = x;
+        if (index === array.length - 1) {
+          this.save(array, callback);
```

Comment:
Isn't `callback` is an optional and can be `undefined` here. Do you need to null-check the `callback!`?